google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java

```diff
@@ -377,6 +451,12 @@ public Builder toBuilder() {
       tsBuilder.setNanos((int) (timestamp % MILLIS_PER_SECOND * NANOS_PER_MILLISECOND));
       builder.setTimestamp(tsBuilder.build());
     }
+    if (receiveTimestamp != null) {
```

Comment:
Optional. Taking into account that this weird time computation logic is copied in two places now, please consider having a private method for that to avoid duplication.